daemon/libnetwork/drivers/macvlan/macvlan_joinleave.go

```diff
@@ -45,9 +47,9 @@ func (d *driver) Join(ctx context.Context, nid, eid string, sboxKey string, jinf
 	}
 	// bind the generated iface name to the endpoint
 	endpoint.srcName = vethName
-	ep := n.endpoint(eid)
-	if ep == nil {
-		return fmt.Errorf("could not find endpoint with id %s", eid)
+	ep, err := n.endpoint(eid)
```

Comment:
Then fetch it again.

Double-checking because before this, the `n.endppoint()` would return a straight `n.endpoints[eid]`, and with the updated code, we use an intermediate var.

(double checking if the previous code would mutate the value in the map, and the new one doesn't, because there's pointers involved and those are always fun to get your head around)